.github/workflows/ci.yml

```diff
@@ -1,44 +1,64 @@
 name: Run CI
+
 on:
   push:
     branches:
       - master
   pull_request:
 
+env:
+  TZ: Asia/Singapore
+  NODE_OPTIONS: --max_old_space_size=8192
+  GENERATE_SOURCEMAP: false
+
 jobs:
-  deploy:
-    name: Run CI
+  deps:
+    name: Install dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup repository
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: yarn
```

Comment:
Hmm you are right! I think the benefit we get from the time savings from removing this step is greater than the additional (free) computing power required for each of the individual jobs in the matrix to re-fetch everything when the cache is invalidated (as opposed to just getting the cache once).

Reply:
Ah, if the reason for this step is so that we only install dependencies once when the lockfile changes, I think this step doesn't exactly do that? Correct me if I'm wrong, but I don't think the `actions/setup-node@v3` action installs dependencies, right? There would need to be a separate step in this job to do that.

Either way, I think removing the `deps` job is a worthy trade-off. We get a simpler configuration at the expense of losing this small optimisation that each lockfile change will only result in a single dependency installation step running.

Reply:
Agreed, I've removed it in 4830a89be75c5846bc06b6dccdb8cbf8f54c6dfd. Thanks!